src/lib.rs

```diff
@@ -33,52 +33,80 @@
 //! -Ten points for Gaston
 //! ```
 
-/// Compare the contents
-#[macro_export]
-macro_rules! assert_contents {
-    ($path:expr, $actual:expr) => {
-        use difference::Changeset;
-        use newline_converter::dos2unix;
-        use std::{env, fs, path::Path};
+use difference::Changeset;
+use newline_converter::dos2unix;
+use std::{env, fs, path::Path};
 
-        let ospath = Path::new($path);
-        let path = &ospath;
-        let var = env::var_os("EXPECTORATE");
-        let overwrite = match var.as_ref().map(|s| s.as_os_str().to_str()) {
-            Some(Some("overwrite")) => true,
-            _ => false,
+#[doc(hidden)]
+// This does the actual heavy lifting while the macro allows panics to appear
+// with the line numbers of the test invoking it. We don't intend consumers of
+// the crate to use this directly.
```

Comment:
> the macro allows panics to appear with the line numbers of the test invoking it

If this is the only reason the panics are done in the macro instead of the function, consider using #\[track_caller\] instead and moving the panics back in line into the logic. https://doc.rust-lang.org/1.50.0/reference/attributes/codegen.html#the-track_caller-attribute

Reply:
Nice! This is much better; thank you.